torch_frame/data/dataset.py

```diff
@@ -360,20 +368,15 @@ def __init__(
             raise ValueError(
                 "Multilabel classification task is not yet supported.")
 
-        self.col_to_sep = canonicalize_col_to_pattern(
-            col_to_sep,
-            [
-                col for col, stype in self.col_to_stype.items()
-                if stype == torch_frame.multicategorical
-            ],
-        )
+        self.col_to_sep = canonicalize_col_to_pattern(col_to_sep, [
+            col for col, stype in self.col_to_stype.items()
+            if stype == torch_frame.multicategorical
+        ], all_inclusive=False)
```

Comment:
If we have multiple multicategorical with a mixture of lists and string as raw data, then we were forced to specify a separator even for the list columns. This pr also changes this, so the specification doesn't need to be all inclusive.